java/lib/ghsl/CommandInjectionRuntimeExec.qll

```diff
@@ -4,124 +4,111 @@ import semmle.code.java.dataflow.DataFlow
 private import semmle.code.java.dataflow.TaintTracking
 import semmle.code.java.dataflow.FlowSources
 
+abstract class Source extends DataFlow::Node {
+  Source() { this = this }
+}
 
-// a static string of an unsafe executable tainting arg 0 of Runtime.exec()
-class ExecTaintConfiguration extends TaintTracking::Configuration {
-    ExecTaintConfiguration() { this = "ExecTaintConfiguration" }
-
-    override
-    predicate
-    isSource(DataFlow::Node source) {
-        source.asExpr() instanceof StringLiteral
-        and source.asExpr().(StringLiteral).getValue() instanceof UnSafeExecutable
+module RuntimeExec {
+  // a static string of an unsafe executable tainting arg 0 of Runtime.exec()
+  module RuntimeExecConfiguration implements DataFlow::ConfigSig {
+    predicate isSource(DataFlow::Node source) {
+      source.asExpr() instanceof StringLiteral and
+      source.asExpr().(StringLiteral).getValue() instanceof UnSafeExecutable
     }
 
-    override
-    predicate
-    isSink(DataFlow::Node sink) {
-        exists(RuntimeExecMethod method, MethodAccess call |
-            call.getMethod() = method
-            and sink.asExpr() = call.getArgument(0)
-            and sink.asExpr().getType() instanceof Array
-        )
+    predicate isSink(DataFlow::Node sink) {
+      exists(RuntimeExecMethod method, MethodCall call |
+        call.getMethod() = method and
+        sink.asExpr() = call.getArgument(0) and
+        sink.asExpr().getType() instanceof Array
+      )
     }
 
-    override
-    predicate
-    isSanitizer(DataFlow::Node node) {
-        node.asExpr().getFile().isSourceFile() and
-        (
-            node instanceof AssignToNonZeroIndex
-            or node instanceof ArrayInitAtNonZeroIndex
-            or node instanceof StreamConcatAtNonZeroIndex
-            or node.getType() instanceof PrimitiveType
-            or node.getType() instanceof BoxedType
-        )
+    predicate isBarrier(DataFlow::Node node) {
+      node.asExpr().getFile().isSourceFile() and
+      (
+        node instanceof AssignToNonZeroIndex or
+        node instanceof ArrayInitAtNonZeroIndex or
+        node instanceof StreamConcatAtNonZeroIndex or
+        node.getType() instanceof PrimitiveType or
+        node.getType() instanceof BoxedType
+      )
     }
-}
+  }
 
-abstract class Source extends DataFlow::Node {
-    Source() {
-        this = this
-    }
-}
+  module RuntimeExecFlow = TaintTracking::Global<RuntimeExecConfiguration>;
 
+  import RuntimeExecFlow::PathGraph
```

Comment:
I think this is not necessary. Only to bring in the edges predicate necessary for path-problems